src/webgpu/api/validation/render_pass/render_pass_descriptor.spec.ts

```diff
@@ -174,6 +176,85 @@ g.test('color_attachments,out_of_bounds')
     t.tryRenderPass(_success, { colorAttachments });
   });
 
+g.test('color_attachments,limits,maxColorAttachmentBytesPerSample,aligned')
+  .desc(
+    `
+    Test that the total bytes per sample of the formats of the color attachments must be no greater
+    than maxColorAttachmentBytesPerSample when the components are aligned (same format).
+  `
+  )
+  .params(u =>
+    u
+      .combine('format', kRenderableColorTextureFormats)
+      .beginSubcases()
+      .combine(
+        'attachmentCount',
+        range(kMaxColorAttachments, i => i + 1)
+      )
+  )
+  .fn(async t => {
+    const { format, attachmentCount } = t.params;
+    const info = kTextureFormatInfo[format];
+
+    const colorAttachments = [];
+    for (let i = 0; i < attachmentCount; i++) {
+      const colorTexture = t.createTexture();
+      colorAttachments.push(t.getColorAttachment(colorTexture));
+    }
+
+    t.tryRenderPass(
+      (info.renderTargetPixelByteCost ?? 0) * attachmentCount <=
```

Comment:
Hmm in `createRenderBundleEncoder.spec.ts` we have at [1]
```
 const shouldError =
      info.renderTargetPixelByteCost === undefined ||
      info.renderTargetPixelByteCost * colorFormatCount >
        t.device.limits.maxColorAttachmentBytesPerSample;
```

` info.renderTargetPixelByteCost === undefined` (snorm format) seems to be an error

But here `info.renderTargetPixelByteCost ?? 0` seems ` info.renderTargetPixelByteCost === undefined` will pass. Feels inconsist here.

Is it because snorm formats are not renderable? but formats are in kRenderableColorTextureFormats.

Reply:
Answering the three related comments here:

I modified the tests to use the same error condition now, but the unrenderable snorm formats are not actually in kRenderableColorTextureFormats. (The typing claims it COULD be, but it's not because of the filter. The typing is a bit misleading). So it didn't matter before w.r.t the correctness of the tests. But I definitely think using the same condition is easier to understand, hence the update.